ocaml/xapi/xapi_vm.ml

```diff
@@ -700,6 +700,49 @@ let set_memory_dynamic_range ~__context ~self ~min ~max =
 	if power_state = `Running
 	then Xapi_xenops.set_memory_dynamic_range ~__context ~self min max
 
+let request_rdp ~__context ~vm ~enabled =
+	let vm_gm = Db.VM.get_guest_metrics ~__context ~self:vm in
+	let vm_gmr = try Some (Db.VM_guest_metrics.get_record_internal ~__context ~self:vm_gm) with _ -> None in  
+	let is_feature_ts_on =
+		match vm_gmr with
+			| None -> false
+			| Some vm_gmr ->
+				let other = vm_gmr.Db_actions.vM_guest_metrics_other in
+				try
+					List.assoc "feature-ts" other = "1"
+				with Not_found -> false
+	in
+	if is_feature_ts_on
+	then
+		Xapi_xenops.request_rdp ~__context ~self:vm enabled
+	else raise (Api_errors.Server_error (Api_errors.vm_lacks_feature_rdp, [ Ref.string_of vm ]))
+
+let request_rdp_on ~__context ~vm =
+	request_rdp ~__context ~vm ~enabled:true
+
+let request_rdp_off ~__context ~vm =
+	request_rdp ~__context ~vm ~enabled:false
+
+(* this is the generic plugin call available to xapi users *)
+let call_plugin ~__context ~vm ~plugin ~fn ~args =
+	if plugin <> "guest-agent-operation" then
+		raise (Api_errors.Server_error(Api_errors.xenapi_missing_plugin, [ plugin ]));
+	let _ = 
+		try List.assoc "foobar" args
+		with _ -> ""
+	in
+
+	match fn with
+	| "request-rdp-on" ->
+		request_rdp_on ~__context ~vm;
+		""
+	| "request-rdp-off" ->
+		request_rdp_off ~__context ~vm;
+		""
+	| value ->
```

Comment:
I would use `_` for the match here instead of `value`, and then `fn` instead of `value` when building `msg` in the next line.
That way is slightly clearer to read, and doesn't rely on `value` being unbound.